compaction/compaction_manager.cc

```diff
@@ -1819,9 +1819,6 @@ future<> compaction_manager::try_perform_cleanup(owned_ranges_ptr sorted_owned_r
     auto& cs = get_compaction_state(&t);
     co_await run_with_compaction_disabled(t, [&] () -> future<> {
         auto update_sstables_cleanup_state = [&] (const sstables::sstable_set& set) -> future<> {
-            // Hold on to the sstable set since it may be overwritten
```

Comment:
"Keeping a shared impl in the sstable set
allows us to update just this set without
requiring rebuild of the whle table's sstable set."

what about full scans? which span all compaction groups. I guess I'll understand it later in the next patches.

Reply:
good point.
We need to deal with that in the compound_sstable_set::for_each_sstable_gently_until override